apps/ejabberd/src/mod_shared_roster_ldap.erl

```diff
@@ -179,13 +179,25 @@ get_jid_info({Subscription, Groups}, User, Server, JID) ->
         error -> {Subscription, Groups}
     end.
 
-in_subscription(Acc, User, Server, JID, Type, _Reason) ->
-    process_subscription(in, User, Server, JID, Type, Acc).
+in_subscription(Acc, _, User, Server, JID, Type, _Reason) ->
+    case process_subscription(in, User, Server, JID, Type) of
+        stop ->
+            {stop, Acc};
+        {stop, false} ->
+            {stop, acc};
```

Comment:
Again, fixed later. About 90% of things you spotted were cleaned up later, but this PR was made from just the first commit, so now I have to apply all those changes here.